src/components/navigation/Tab.tsx

```diff
@@ -0,0 +1,77 @@
+import classnames from 'classnames';
+import type { JSX } from 'preact';
+
+import type { PresentationalProps } from '../../types';
+import { downcastRef } from '../../util/typing';
+
+import ButtonBase from '../input/ButtonBase';
+
+type ComponentProps = {
+  /**
+   * Text string representing the content of the tab when selected. The tab
+   * button will be sized to accommodate this string in bold text. This can
+   * prevent tab jiggle.
+   */
+  textContent?: string;
+  selected?: boolean;
+  variant?: 'basic';
+};
+
+type HTMLAttributes = Omit<
+  JSX.HTMLAttributes<HTMLButtonElement>,
+  'size' | 'icon' | 'title'
+>;
+
+export type TabProps = PresentationalProps & ComponentProps & HTMLAttributes;
+
+/**
+ * Render a button with appropriate ARIA tab affordances
+ */
+const TabNext = function Tab({
+  children,
+  classes,
+  elementRef,
+
+  textContent,
+  selected = false,
+  variant = 'basic',
+
+  ...htmlAttributes
+}: TabProps) {
+  return (
+    <ButtonBase
+      {...htmlAttributes}
+      classes={classnames(
+        'enabled:hover:text-brand-dark',
+        {
+          'font-bold': selected && variant === 'basic',
+        },
+        classes
+      )}
+      elementRef={downcastRef(elementRef)}
+      aria-selected={selected}
+      role="tab"
+      data-component="Tab"
+    >
+      <span
+        data-content={textContent}
+        data-testid="sizing-wrapper"
+        className={classnames({
+          // Set the content of this span's ::before pseudo-element to
+          // `textContent` and make it bold.
+          'before:content-[attr(data-content)] before:font-bold': textContent,
+          // Make the ::before occupy space within the button, but make it
+          // invisible. This ensures that the tab button is wide enough to show
+          // bolded text even if the visible text is not currently bold. See
+          // https://css-tricks.com/bold-on-hover-without-the-layout-shift/
```

Comment:
The buttons at the top of this interface itself use this trick so that they don't move around when active:

<img width="1153" alt="image" src="https://user-images.githubusercontent.com/439947/215200041-a0e80fbf-3525-42cf-81c0-024b0a76b647.png">

Reply:
Clever! 😃